### PR TITLE
Port from JsonCPP to Boost.Json for json parsing and serializing

### PR DESCRIPTION
**Pull request by RyeMutt**
This ports the viewer to Boost.Json to eliminate the now-unneeded dependency on JsonCPP 

This depends on an update to Boost 1.84+

### CODE REVIEW

**Comment by bennettgoble** [comment]
The build here failed as expected, we need to incorporate a version of boost with the json module: https://github.com/secondlife/viewer/issues/1057

**Comment by marchcat** [comment]
Thanks!
Let's try to get this into Maint B (https://github.com/secondlife/viewer/tree/release/maint-b) along with #1057.

**Comment by bennettgoble** [comment]
> Thanks! Let's try to get this into Maint B (`release/maint-b`) along with #1057.

Thanks, @marchcat, @RyeMutt . We'll need to get the boost library and dependent 3p packages updated before this is merged. Also, it will be worth confirming with folks that the strategy here is what we want to go with (I think it is, but we just need to confirm.)

Rationale:

- **boost.json** performance seems decent, from these benchmarks: https://230.jsondocs.prtest.cppalliance.org/libs/json/doc/html/json/benchmarks.html (comparable to rapidjson).
- Adopting this change will eliminate a dependency (jsoncpp)

**Comment by marchcat** [comment]
Retargeting to Maint A to follow #1097.

**Review by marchcat — APPROVED** [review]
The code looks good, but let's check the build first.

**Comment by marchcat** [comment]
Windows build is fine, but mac [fails](https://github.com/secondlife/viewer/actions/runs/8505389386/job/23293782267#step:10:2264).

I'll look deeper into that issue later today.

**Comment by RyeMutt** [comment]
I have additional changes to clean up some exception safety issues that'll be pushed later today

**Comment by RyeMutt** [comment]
Should be good to go now! Cleaned up some exception safety issues around lltranslate by changing to the non-throwing api.

**Review by bennettgoble — COMMENTED** [review]
Is there any reason to leave JsonCPP in autobuild.xml?

File: indra/newview/llappviewerwin32.cpp [hunk 1]
```diff
@@ -722,24 +722,25 @@ bool LLAppViewerWin32::init()
         }
         else
         {
-            Json::Reader reader;
-            Json::Value build_data;
-            if (!reader.parse(inf, build_data, false)) // don't collect comments
+            boost::json::error_code ec;
+            boost::json::value build_data = boost::json::parse(bas, ec);
```
Comment (marchcat) [inline indra/newview/llappviewerwin32.cpp:726]: Github build throws `llappviewerwin32.cpp(726,64): error C2065: 'bas': undeclared identifier` here. However, my local Windows build is fine.

Comment (RyeMutt) [reply indra/newview/llappviewerwin32.cpp]: Should be fixed with the next commit!

**Comment by RyeMutt** [comment]
Nope, just missed it during my cleanup pass.

**Comment by bennettgoble** [comment]
> Nope, just missed it during my cleanup pass.

@RyeMutt let's get the autobuild.xml file updated (remove jsoncpp) then - @marchcat, is there anything else outstanding?

**Comment by marchcat** [comment]
So far, no. Let's test it.

**Comment by RyeMutt** [comment]
Cleaned up the remainders of jsoncpp from autobuild.xml and cmake

**Comment by marchcat** [comment]
Thank you!

**Comment by bennettgoble** [comment]
Windows Release build from https://github.com/secondlife/viewer/actions/runs/8576215987 seems to run fine.

**Comment by marchcat** [comment]
Awesome, merging.

**Pull request merged**